python/ray/tune/schedulers/hb_bohb.py

```diff
@@ -140,7 +140,10 @@ def choose_trial_to_run(
             scrubbed = [b for b in hyperband if b is not None]
             for bracket in scrubbed:
                 for trial in bracket.current_trials():
-                    if trial.status == Trial.PENDING:
+                    if (
+                        trial.status == Trial.PAUSED
+                        and trial in bracket.trials_to_unpause
```

Comment:
This is a change in the hyperband - instead of directly setting the trial status to PENDING, we add them to the `trials_to_unpause` set. Then the scheduler will selet them in `choose_trial_to_run`

Reply:
nice I like this.